x-pack/platform/plugins/shared/embeddable_alerts_table/public/components/config_editor_flyout.tsx

```diff
@@ -0,0 +1,242 @@
+import { isEmptyExpression } from '@kbn/response-ops-alerts-filters-form/utils/filters';
+import {
+  getAvailableSolutions,
+  getRuleTypeIdsForSolution,
+} from '@kbn/response-ops-alerts-filters-form/utils/solutions';
+import { useGetInternalRuleTypesQuery } from '@kbn/response-ops-rules-apis/hooks/use_get_internal_rule_types_query';
+import type { AlertsFiltersExpression } from '@kbn/response-ops-alerts-filters-form/types';
+import type { RuleTypeSolution } from '@kbn/alerting-types';
+import type { EuiSuperSelect } from '@elastic/eui/src/components/form/super_select/super_select';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import type { EmbeddableAlertsTableConfig } from '../types';
+import {
+  CONFIG_EDITOR_ADD_TABLE_TITLE,
+  CONFIG_EDITOR_CANNOT_LOAD_RULE_TYPES_DESCRIPTION,
+  CONFIG_EDITOR_CANNOT_LOAD_RULE_TYPES_TITLE,
+  CONFIG_EDITOR_EDITOR_CLOSE_LABEL,
+  CONFIG_EDITOR_EDITOR_SAVE_LABEL,
+  CONFIG_EDITOR_EDIT_TABLE_TITLE,
+  CONFIG_EDITOR_FILTERS_FORM_TITLE,
+  SWITCH_SOLUTION_CONFIRM_CANCEL_MESSAGE,
+  SWITCH_SOLUTION_CONFIRM_CONFIRM_MESSAGE,
+  SWITCH_SOLUTION_CONFIRM_MESSAGE,
+  SWITCH_SOLUTION_CONFIRM_TITLE,
+  CONFIG_EDITOR_CLEAR_FILTERS_LABEL,
+} from '../translations';
+
+export interface ConfigEditorFlyoutProps {
+  initialConfig?: EmbeddableAlertsTableConfig;
+  onSave: (newConfig: EmbeddableAlertsTableConfig) => void;
+  onCancel: () => void;
+  services: {
+    http: CoreStart['http'];
+    notifications: CoreStart['notifications'];
+    overlays: CoreStart['overlays'];
+  };
+}
+
+const EMPTY_FILTERS: AlertsFiltersExpression = [{ filter: {} }];
+
+export const ConfigEditorFlyout = ({
+  initialConfig,
+  onSave,
+  onCancel,
+  services,
+}: ConfigEditorFlyoutProps) => {
+  const { http, overlays } = services;
+  const flyoutBodyRef = useRef<HTMLDivElement>(null);
+  const solutionSelectorRef = useRef<EuiSuperSelect<RuleTypeSolution>>(null);
+  const [filters, setFilters] = useState<
+    EmbeddableAlertsTableConfig['query']['filters'] | undefined
+  >(initialConfig?.query?.filters ?? EMPTY_FILTERS);
+  const formRef = useRef<AlertsFiltersFormHandle>(null);
+  const {
+    data: ruleTypes,
+    isLoading: isLoadingRuleTypes,
+    isError: cannotLoadRuleTypes,
+  } = useGetInternalRuleTypesQuery({ http });
+  const availableSolutions = useMemo(
+    () => (ruleTypes ? getAvailableSolutions(ruleTypes) : undefined),
+    [ruleTypes]
+  );
+  const [solution, setSolution] = useState<EmbeddableAlertsTableConfig['solution'] | undefined>(
+    initialConfig?.solution ??
+      (availableSolutions && availableSolutions.length === 1 ? availableSolutions[0] : undefined)
+  );
+  const ruleTypeIds = useMemo(
+    () => (!ruleTypes || !solution ? [] : getRuleTypeIdsForSolution(ruleTypes, solution)),
+    [ruleTypes, solution]
+  );
+
+  const resetFilters = useCallback(() => {
+    setFilters(EMPTY_FILTERS);
+  }, []);
+
+  const onSolutionChange = useCallback(
+    (newSolution: RuleTypeSolution) => {
```

Comment:
This adds a bit to the component length... I considered extracting it, but since it relies on quite a few local variables, I wasn’t sure the tradeoff would be worth it. Happy to revisit if you think it’s worth moving out